### PR TITLE
update as-transformer documentation

### LINKED ISSUE #117: as-transformer: Enhance or add to code documentation: classes, functions, etc.
- [ ] #189

### PR DESCRIPTION
**Pull request by Elli610**
closes #117

### CODE REVIEW

File: packages/as-transformer/src/transformers/file2ByteArray.ts [hunk 1]
```diff
@@ -9,14 +9,20 @@ import { SimpleParser } from 'visitor-as';
...
- * File2ByteArray transformer
+ * The File2ByteArray transformer object allows to transform writes a function call which aims to transform
+ * an ast into a byteArray.
+ *
  */
 export class File2ByteArray {
   static strPattern = 'fileToByteArray';
 
   /**
+   * Transforms a {@link CallExpression} into a {@link Expression} by replacing the call with the new expression.
```
Comment (0xSwapFeeder) [inline packages/as-transformer/src/transformers/file2ByteArray.ts:20]:
```suggestion
   * Transforms a 'fileToByteArray' **call expression** into an **expression** to extract the file as a byte array.
```

File: packages/as-transformer/src/transformers/file2ByteArray.ts [hunk 2]
```diff
@@ -9,14 +9,20 @@ import { SimpleParser } from 'visitor-as';
 import * as fs from 'fs';
 
 /**
- * File2ByteArray transformer
+ * The File2ByteArray transformer object allows to transform writes a function call which aims to transform
```
Comment (0xSwapFeeder) [inline packages/as-transformer/src/transformers/file2ByteArray.ts:12]: That's very unclear, you may not want to precise what a transformer is again here.
Just talk about the goal of this specific transformer
(PS: It doesn't change an AST into a byte array, it creates the following **expression**: 'transforms a file into a byte array'  and stores it into the AST as a **new AST Node** replacing the old **call expression** node)

Comment (0xSwapFeeder) [reply packages/as-transformer/src/transformers/file2ByteArray.ts:12]: @gregLibert tell me if i'm right

File: packages/as-transformer/src/transformer.ts [hunk 3]
```diff
@@ -10,7 +10,17 @@ import { TestTable } from './transformers/testTable.js';
 
 const callTransformers = [File2ByteArray, TestTable];
 
+/**
+ * The Transformer class allows you to parse the {@link CallExpression} to look for a specific method and
+ * update it as you need.
+ */
 export class Transformer extends TransformVisitor {
+  /**
+   * Looks for a specific method and replaces it with the new expression.
```
Comment (0xSwapFeeder) [inline packages/as-transformer/src/transformer.ts:19]:
```suggestion
   * Looks for a given **call expression** in the AST Nodes and tries to replace it with the new **expression** node with the registered custom transformers.
```

File: packages/as-transformer/src/transformer.ts [hunk 4]
```diff
@@ -23,6 +33,11 @@ export class Transformer extends TransformVisitor {
     return super.visitCallExpression(node);
   }
 
+  /**
+   * Visits the sources and replaces the {@link CallExpression} with the new expression.
```
Comment (0xSwapFeeder) [inline packages/as-transformer/src/transformer.ts:37]: @gregLibert not sure about this method, is it the method called after AST generation used to apply customized AST operations on targeted sources before compile time ?

**Review by 0xSwapFeeder — CHANGES_REQUESTED** [review]
Questions raised.

File: packages/as-transformer/src/transformers/file2ByteArray.ts (hunk 2 above)
Comment (gregLibert) [reply packages/as-transformer/src/transformers/file2ByteArray.ts:12]: This tranformer is updating the AST by:
- replacing the node containing a call to the fileToByteArray function having a file as argument
- by a node containing an array of bytes corresponding to the content of the file given as argument.
Does it make sens :thinking:

File: packages/as-transformer/src/transformer.ts (hunk 4 above)
Comment (gregLibert) [reply packages/as-transformer/src/transformer.ts:38]: it's one of the three transformer hooks as defined [here](https://www.assemblyscript.org/compiler.html#transforms). It is the first one, called just after parsing the code

File: packages/as-transformer/src/transformer.ts [hunk 5]
```diff
@@ -23,6 +34,14 @@ export class Transformer extends TransformVisitor {
...
+   *
+   * @privateRemarks
+   * This is one of the three transformer hooks as defined here. It is the first one, called just after parsing the cod
```
Comment (0xSwapFeeder) [inline packages/as-transformer/src/transformer.ts:41]: Well, need rephrasing for documentation and a link for 'here'

**Review by 0xSwapFeeder — CHANGES_REQUESTED** [review]

File: packages/as-types/assembly/amount.ts [hunk 6]
```diff
@@ -1,202 +0,0 @@
...
-   *
-   * @param other - Amount to check against.
-   *
-   * @returns true if the amounts are different.
-   */
-  @operator('!=')
-  notEqual(other: Amount): boolean {
-    return !(this == other);
-  }
-}
```
Comment (Ben-Rey) [inline packages/as-types/assembly/amount.ts:202]: @Elli610 Why did you remove the content of this file? same for Result

Comment (Elli610) [reply packages/as-types/assembly/amount.ts:202]: I didn't. It wasn't deleted before the rebase

File: packages/as-transformer/src/transformer.ts [hunk 7]
```diff
@@ -19,7 +19,19 @@ const callTransformers = [File2ByteArray, TestTable];
...
 
+/**
+ * The `Transformer` class extends the `TransformVisitor` class from visitor-as and overrides its methods to perform
+ * custom transformations on the AST during the compilation process.
+ * It looks for specific function calls or decorators and replaces them with new expressions
+ * or functions as needed.
+ */
 export class Transformer extends TransformVisitor {
+  /**
+   * Checks if the function declaration node has a specific decorator, and if so, transforms it.
```
Comment (0xSwapFeeder) [inline packages/as-transformer/src/transformer.ts:30]: Note sure that is the only goal of this function, I mean we use it that way for the moment but it does not transform the function, it only generates data from it.

**Review by 0xSwapFeeder — CHANGES_REQUESTED** [review]

File: packages/as-transformer/src/transformer.ts (hunk 7 above)
Comment (Ben-Rey) [reply packages/as-transformer/src/transformer.ts:30]: Sorry not my scope, I'll remove it

**Review by Ben-Rey — APPROVED** [review]

**Review by gregLibert — APPROVED** [review]

**Review by 0xSwapFeeder — APPROVED** [review]
